app/api/auth/create_access_token.go

```diff
@@ -1,119 +1,169 @@
+// summary: Create or refresh an access token
+// description:
+//     If the "Authorization" header is not given, the service converts the given code into tokens,
+//     creates or updates the authenticated user in the DB with the data returned by the login module,
+//     and saves new access & refresh tokens into the DB as well.
+//
+//
+//     If the "Authorization" header is given, the service refreshes the access token
+//     (locally for temporary users or via the login module for normal users) and
+//     saves it into the DB keeping only the input token (from authorization headers) and the new token.
+//     Since the login module responds with both access and refresh tokens, the service updates the user's
+//     refresh token in this case as well.
+//
+//
+//   * One of the “Authorization” header and the '{code}' parameter should be present (not both at once).
+// security: []
+// parameters:
+// - name: code
+//   in: query
+//   description: OAuth2 code
+//   type: string
+// - name: code_verifier
+//   in: query
+//   description: OAuth2 PKCE code verifier
+//   type: string
 // responses:
 //   "201":
 //     description: "Created. Success response with the new access token"
 //     in: body
 //     schema:
 //       "$ref": "#/definitions/userCreateTmpResponse"
-//   "401":
-//     "$ref": "#/responses/unauthorizedResponse"
+//   "400":
+//     "$ref": "#/responses/badRequestResponse"
 //   "500":
 //     "$ref": "#/responses/internalErrorResponse"
 func (srv *Service) createAccessToken(w http.ResponseWriter, r *http.Request) service.APIError {
-	user := srv.GetUser(r)
-	oldAccessToken := auth.BearerTokenFromContext(r.Context())
-
-	var newToken string
-	var expiresIn int32
-
-	if user.IsTempUser {
-		service.MustNotBeError(srv.Store.InTransaction(func(store *database.DataStore) error {
-			sessionStore := store.Sessions()
-			// delete all the user's access tokens keeping the input token only
-			service.MustNotBeError(sessionStore.Delete("user_id = ? AND access_token != ?",
-				user.GroupID, oldAccessToken).Error())
-			var err error
-			newToken, expiresIn, err = auth.CreateNewTempSession(sessionStore, user.GroupID)
-			return err
-		}))
-	} else {
-		// We should not allow concurrency in this part because the login module generates not only
-		// a new access token, but also a new refresh token and revokes the old one. We want to prevent
-		// usage of the old refresh token for that reason.
-		service.MustNotBeError(userIDsInProgress.withLock(user.GroupID, r, func() error {
-			newToken, expiresIn = srv.refreshTokens(r.Context(), user, oldAccessToken)
-			return nil
-		}))
+	if len(r.Header["Authorization"]) != 0 {
+		if len(r.URL.Query()["code"]) != 0 {
+			return service.ErrInvalidRequest(
+				errors.New("only one of the 'code' query parameter and the 'Authorization' header can be given"))
+		}
+		auth.UserMiddleware(srv.Store.Sessions())(service.AppHandler(srv.refreshAccessToken)).ServeHTTP(w, r)
```

Comment:
you should put a comment to summarize in which case you are at the end on this line:
> // "Authorization" header given, requesting a new token from the given token

Reply:
fixed